R/string_operation.R

```diff
@@ -366,23 +366,31 @@ do_tokenize <- function(df, input, token = "words", keep_cols = FALSE,  drop = T
   ret
 }
 
-#' Get idf for terms
-calc_idf <- function(group, term, smooth_idf = FALSE){
+#' Input:
+#' group - vector of documents
+#' term - vector terms with the same length as group.
+#' tf - vector of tf with the same length as group. It can be after weight function is applied,
+#'      in which case, we assume that group-term combination in the input is unique, and summing up at the sparseMatrix would not happen. 
+#' count_per_doc - vector of word count per doc. It is tf before weight function is applied.
+#' Output: data.frame of document frequency and tf-idf. The number of rows are the same as the input vectors.
+calc_tfidf <- function(group, term, tf, count_per_doc, smooth_idf = FALSE){
   loadNamespace("Matrix")
   loadNamespace("text2vec")
   if(length(group)!=length(term)){
     stop("length of document and terms have to be the same")
   }
   doc_fact <- as.factor(group)
   term_fact <- as.factor(term)
-  sparseMat <- Matrix::sparseMatrix(i = as.numeric(doc_fact), j = as.numeric(term_fact))
+  sparseMat <- Matrix::sparseMatrix(i = as.numeric(doc_fact), j = as.numeric(term_fact), x = tf) # With weight function applied. To calculate tfidf.
```

Comment:
since all the other variables are snake_case, it might be better to use snake_case like `sparce_mat` ?

Reply:
Same for sparseMatForCount. Maybe `sparse_mat_for_count` ?

Reply:
Adjusted the variable name style.